src/shex-wrapper.js

```diff
@@ -0,0 +1,129 @@
+// shex-wrapper.js
+
+var _ = require('underscore');
+
+var util = require('util');
+var logger = require('./logger');
+var createLm = require('./create-lm');
+var createState = require('./create-state');
+var factory = require('./pipeline-component-factory');
+var wrapperHelper = require('./wrapper-helper');
+
+var N3 = require("n3");
+var jsonld = require("jsonld").promises;
+var shexiface = require("../shex/shexiface");
+
+/**
+ * This module provides a Shex wrapper.  It is responsible for setting up the
+ * component, and then invoking the RDF pipeline factory to complete building
+ * the component and setting up the vni and state metadata.
+ *
+ * @param nodeDef
+ *
+ * @return a promise to create the noflo rdf component
+ */
+module.exports = function(nodeDef) {
+    return factory({
+        inPorts: _.isArray(nodeDef.inPorts) ?
+            nodeDef.inPorts.reduce(function(map, obj) {
+                if (_.isObject(obj)) {
+                    return obj;
+                }
+                return {[obj]: {dataType: obj == 'input' ? 'all' : 'string'}};
+            }, {}) : nodeDef.inPorts || {
+                input: {
+                    dataType: 'all'
+                }
+            }
+    }, {
+        fRunUpdater: _.partial(fRunUpdater, nodeDef)
+    });
+};
+
+/**
+ * RDF Pipeline SHEX wrapper fRunUpdater API as documented here:
+ *    https://github.com/rdf-pipeline/noflo-rdf-pipeline/wiki/Wrapper-API
```

Comment:
Please add more comment saying what it expects on input, and what it does (with the conversions between JSON and RDF).